### ISSUE TITLE
Create real test demo

### ISSUE THREAD

**Issue by anhtuank7c** [ChuckHeintzelman/l5beauty#3]
Dear Chuck Heintzelman.
Can you write a real demo phpunit test at this project?
And how we integrate ACL permission?

How to control test in complex project.
Ex:
- Create test create a new post, delete post.
- Manage user, role, permission.

Many thanks

**Comment by ChuckHeintzelman** [comment]
Sorry I have several deadlines and don’t have the spare time.

Jeffrey Way wrote a book “Laravel Testing Decode” that, although originally written for Laravel 4, fits well with Laravel 5.

Cheers,
— Chuck

From:  Nguyễn Anh Tuấn
Reply-To:  ChuckHeintzelman/l5beauty
Date:  Wednesday, September 2, 2015 at 1:45 AM
To:  ChuckHeintzelman/l5beauty
Subject:  [l5beauty] Create real test demo (#3)

Dear Chuck Heintzelman.
Can you write a real demo phpunit test at this project?
How to control test in complex project.
Ex: test create a new post, delete post.
Many thanks

**Comment by nguyenphuocnhatthanh** [comment]
You can use package roles or entrust for ACL

**Comment by anhtuank7c** [comment]
Roles at this link "https://github.com/romanbican/roles" right? @nguyenphuocnhatthanh

**Comment by nguyenphuocnhatthanh** [comment]
yep :)

**Comment by anhtuank7c** [comment]
@nguyenphuocnhatthanh thank you.
@ChuckHeintzelman When you have plan, please help write real phpunit test.
Many Thanks :dancer: 
Have a good day.

**Comment by peterbata** [comment]
I read sections of your book and downloaded the master source code. There appears to be errors in the code. If my memory serves me right it may have been in the routes.php file.

**Comment by ChuckHeintzelman** [comment]
Yeah, the book is for Laravel 5.1 and the upgrades to Laravel 5.2 have some errors.

Sounds like you figured it out. Great.

I’ll be releasing an update to the book for Laravel 5.2 in the near future.

Cheers,
— Chuck
http://BundleRabbit.com
Promote your ebooks in DIY Bundles
Get more ebooks for less money

On March 17, 2016 at 6:01:57 AM, Peter Bata (notifications@github.com) wrote:

I read sections of your book and downloaded the master source code. There appears to be errors in the code. If my memory serves me right it may have been in the routes.php file.

**Comment by peterbata** [comment]
Thank you so much for the response Chuck. I have been struggling with this for days and am glad that you helped confirm that I wasn't losing my marbles. Looking forward to the 5.2 version. Have a wonderful day.

Peter

**Comment by ChuckHeintzelman** [comment]
It may be helpful to edit your composer.json file and change the “laravel/framework”: “5.2._” to “5.1._"

— Chuck
http://BundleRabbit.com
Promote your ebooks in DIY Bundles
Get more ebooks for less money

On March 17, 2016 at 6:13:45 AM, Peter Bata (notifications@github.com) wrote:

Thank you so much for the response Chuck. I have been struggling with this for days and am glad that you helped confirm that I wasn't losing my marbles. Looking forward to the 5.2 version. Have a wonderful day.

Peter

**Comment by peterbata** [comment]
I'll try that. Would that explain why I can't seem to get past a RouteCollection.php error at line 161. If my memory serves me right. Not to worry, I'll see if the version change helps

Thanks again

Peter

**Comment by ChuckHeintzelman** [comment]
Yeah, Laravel 5.2 changed the routing structure to allow different groups of middleware.

What that means is you could have a group of middleware for one thing (like web requests) and a different group for another (like api requests).

A cool feature, but backwards breaking.

— Chuck
http://BundleRabbit.com
Promote your ebooks in DIY Bundles
Get more ebooks for less money

On March 17, 2016 at 6:19:36 AM, Peter Bata (notifications@github.com) wrote:

I'll try that. Would that explain why I can't seem to get past a RouteCollection.php error at line 161. If my memory serves me right. Not to worry, I'll see if the version change helps

Thanks again

Peter

**Comment by peterbata** [comment]
Hello again Chuck,

I went ahead and pulled in Laravel Framework version 5.1.31 (LTS)

I made the appropriate changes to my .env file.

When launching 

1)    php artisan serve followed by

2)    http://localhost:8000

I get the following error:

I did run php artisan key:generate and made sure to enter that key info into config/app.php (It was already in my .env file)

No change – Same error

Thank you for your time and assistance.

Peter

**Comment by ChuckHeintzelman** [comment]
Ah, what the problem probably is that now the framework is version 5.1._, but the application installed is still 5.2._.

I haven’t done this before, but when you create the project, let’s say it’s called “myapp”, try the following:

composer create-project laravel/laravel myapp 5.1.*

Like I said, I haven’t tried that before, but it might just do the trick.

Cheers,
— Chuck

On March 17, 2016 at 1:47:01 PM, Peter Bata (notifications@github.com) wrote:

Hello again Chuck,

I went ahead and pulled in Laravel Framework version 5.1.31 (LTS)

I made the appropriate changes to my .env file.

When launching

1) php artisan serve followed by

2) http://localhost:8000

I get the following error:

I did run php artisan key:generate and made sure to enter that key info into config/app.php (It was already in my .env file)

No change – Same error

Thank you for your time and assistance.

Peter

**Comment by peterbata** [comment]
Once again Chuck, I really do appreciate the prompt response. And, my aplogies for burdening you with this issue.

Actually, this is what I did.

1)    I changed into a directory that I created called “ l5test”where I had extracted your l5beauty-master files.

2)    I ran composer init, chose the laravel/laravel package but specified version 5.1 as can be seen in the results below

3)    Composer init created a very clean composer.json file. See below.

4)    I just can’t see what the problem might be.

5)    Have a yourself a great evening and a wonderful Friday. Peter

[cid:image001.jpg@01D1809C.75D4D130]

[cid:image002.jpg@01D1809C.EE758DF0]

**Comment by ChuckHeintzelman** [comment]
Peter,

Sorry I’ve been super-slammed at work. I have started the revisions to the L5 Beauty tutorial for Laravel 5.2. So hopefully when that’s wrapped it’ll help with much of these issues.

On March 20, 2016 at 3:07:52 PM, Peter Bata (notifications@github.com) wrote:

Once again Chuck, I really do appreciate the prompt response. And, my aplogies for burdening you with this issue.

Actually, this is what I did.

1) I changed into a directory that I created called “ l5test”where I had extracted your l5beauty-master files.

2) I ran composer init, chose the laravel/laravel package but specified version 5.1 as can be seen in the results below

3) Composer init created a very clean composer.json file. See below.

4) I just can’t see what the problem might be.

5) Have a yourself a great evening and a wonderful Friday. Peter

[cid:image001.jpg@01D1809C.75D4D130]

[cid:image002.jpg@01D1809C.EE758DF0]

**Comment by chan2119** [comment]
Hello Chuck 

I have a question in this tutorial,

when I save new post then it show me like this

"FatalThrowableError in Markdowner.php line 15:
Fatal error: Class 'Michelf\SmartyPants' not found"

but I have already  install SmartyPants by composer , and the composer.json was updated.

so I can't  figure out why.

when you feel free , could you please help me fix it.
thank you so much!

**Comment by ChuckHeintzelman** [comment]
A couple things to check:
1) Triple check there's no typos. Often a FatalThrowableError occurs because of that
2) Make sure you do a "composer update" after modifying the composer.json file

If that doesn't fix it, I'm a bit flummoxed. SmartyPants adds the curly quotes so you could get by without it, just comment out that line.

**Comment by mpmurph** [comment]
@chan2119 I was having exactly the same problem and discovered two things: first, the namespace for SmartyPants is actually michelf not Michelf. Also, for whatever reason, SmartyPants does not have an autoload in its composer.json file. I was able to get around the problem by putting the following in my app's composer.json file: 

```
"michelf\\": "vendor/michelf/php-smartypants/michelf"
```

Then I put the following at the top of my Markdowner class:

```
use michelf\SmartyPants;
```

Hope that helps!

**Comment by mpmurph** [comment]
@chan2119 Follow-up - although the class is now recognized by my app, I am getting all sorts of errors about undefined constants.

**Comment by ChuckHeintzelman** [comment]
Try using "michelf/php-smartypants": "dev-lib" in composer.json instead of the beta label.

Be sure to "composer update" afterwards

**Comment by mpmurph** [comment]
@ChuckHeintzelman @chan2119 That fixed everything! The tests are all running with a green light now and the namespace is back to Michelf so I have been able to delete the autoload statement from my composer.json file and switch back to:

```
use \Michelf\SmartyPants;
```

Thank you, Chuck, for the quick follow-up! I had been knocking my head against a wall trying to get everything to work - on the plus side, I learned more about psr-4 syntax. Now, I am primed to continue along with your tutorial... Thanks again for the help!
